src/Elements/types.jl

```diff
@@ -0,0 +1,180 @@
+const HEADERS   = ["#", "##", "###", "####", "#####", "######"]
+const EMPHASIS  = ["*", "**"]
+
+const CATEGORIES = [:module,    :function, :method, :macro, :type,
+                    :typealias, :global,   :symbol, :tuple, :aside]
+
+
+## Config Nodes
+abstract ConfigNode
+
+type Config <: ConfigNode
+    objfilter               :: Any
+    style_obj               :: ASCIIString
+
+    const defaults = Dict{Symbol, Any}(
+        :objfilter                  => CATEGORIES,
+        :style_obj                  => "####",
```

Comment:
> It would remove the need for
> ```
> const HEADERS   = ["#", "##", "###", "####", "#####", "######"]
>
> const EMPHASIS  = ["*", "**"]
> ```

If you think - a function should also be fine.

**flexible**: one reason I introduced in the previous version the `title(), header() ect.` was that user could have added custome types and rendermacros so if one wanted something special: they could just hook it in similar to Docile metamacros.

Why don't we leave for the moment the simple string and see how all the other stuff comes together so that we can work on more details when we have something at least producing output as good as the current version.